src/utils.js

```diff
@@ -450,11 +450,33 @@ nv.utils.calcTicksX = function(numTicks, data) {
 
 
 /*
-returns number of ticks to actually use on Y axis, based on chart data
-*/
-nv.utils.calcTicksY = function(numTicks, data) {
-    // currently uses the same logic but we can adjust here if needed later
+ returns number of ticks to actually use on Y axis, based on chart data
+ */
+nv.utils.calcTicksY = function(numTicks, data, yAccessor) {
+  if (yAccessor) {
+    // find max number of values from all data streams
+    var numValues = 1;
+    for (var i=0; i < data.length; i += 1) {
+      var values = data[i] && data[i].values ? data[i].values : [];
+      var maxValue;
+      for (var j=0; j < values.length; j += 1) {
+        maxValue = values[j] && values[j][yAccessor] ? values[j][yAccessor]: 0;
+        numValues = maxValue > numValues ? maxValue : numValues;
+      }
+    }
```

Comment:
This seems super inefficient... so this is looking at every node on the chart?    Surely there is a better way to chose a tick count...

Reply:
Let's see. Incoming `data` for the `discreteBarChart` always has the following format:
```
[
  {
    "values": [
      {
        "label": "Label1",
        "value": 3,
        "series": 0
      },
      {
        "label": "Label2",
        "value": 123,
        "series": 0
      },
      {
        "label": "Label3",
        "value": 777,
        "series": 0
      },
        ...
    ]
  }
]
```
So complexity would be `O(n)` where `n = data[0].values.length`. The linear complexity is obviously efficient and very fast.

Reply:
Just to clarify, the idea behind this pull request is to calculate y axis ticks for the discrete bar chart based on y values, not on x values as it is now https://jsfiddle.net/ovvn/g5qwwthx/1/embedded/result/. We've been using the fix for half a year on production and it works great for us.

Reply:
Yes, it's O(n), but for very large (millions) of points n is still sufficiently big to potentially cause problems in a browser.  And that's not just worst case, it's every time because you always have to traverse the entire array.  I see you're thinking about a simple bar chart, but this function is used by more than just that... if you want to override the function to calculate the tick count  just for one type of chart, then I'd keep all the logic in the chart itself.  

Also, what's the difference between getY and yAccessor options?  Seems like that's overlapping functionality?

Reply:
Traversal in an array with millions of items is not a problem. What can be a problem is appending millions of elements into a browser's DOM, but that has nothing to do with the pull request. The functionality is not made for just one type of chart, it's generic and can be extended for other types as well. The difference between getY and yAccessor is that `getY` is a function and `yAccessor` is a string (y value property name). But that's a good point, don't remember why I made the property name to be passed explicitly. Think we can reuse the `getY` fn.